Update StableMIR doc to reflect current status

celinval opened the pull request:
Update stable-mir documentation, since we no longer use git subtree, and we have 2 different crates.

<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r\? <reviewer name> (with the `\` removed)
-->
<!-- homu-ignore:end -->

celinval committed:
Update StableMIR doc to reflect current status

We no longer use git subtree, and we have 2 different crates.

compiler-errors commented on compiler/stable_mir/README.md:
```diff
@@ -1,93 +1,25 @@
-
-Then open a PR against rustc just like a regular PR.
+Our goal is to start publishing `stable_mir` into crates.io.
+Until then, users will use this as any other rustc crate, via extern crate.
```
Perhaps mention that users need to install the `rustc-dev` component.

compiler-errors commented on compiler/stable_mir/README.md:
```diff
@@ -1,93 +1,25 @@
 
 ## Stable MIR Design
 
 The stable-mir will follow a similar approach to proc-macro2. It’s
```
```suggestion
The stable-mir will follow a similar approach to proc-macro2. Its
```

compiler-errors commented on compiler/stable_mir/README.md:
```diff
@@ -1,93 +1,25 @@
 
 The stable-mir will follow a similar approach to proc-macro2. It’s
-implementation will eventually be broken down into two main crates:
+implementation is done using two main crates:
```
```suggestion
implementation is split between two main crates:
```

compiler-errors commented on compiler/stable_mir/README.md:
```diff
@@ -1,93 +1,25 @@
-implementation will eventually be broken down into two main crates:
+implementation is done using two main crates:
 
 - `stable_mir`: Public crate, to be published on crates.io, which will contain
```
```
- `stable_mir`: Public crate, to be published on crates.io, whics contains
the stable data structures and APIs to access them. It calls to `rustc_smir`.
- `rustc_smir`: This crate serves as `stable_mir`'s interface to the compiler.
It is responsible for gathering all the information requested, and converting
them to their stable data structures.
```

compiler-errors replied:
Github didn't allow me to rewrite this as an inline suggestion, but this section seems wrong. AFAICT, the `rustc_smir` crate is what does the conversion; hence it having all the `.stable` calls.

celinval replied:
Indeed. Not for long though, but I'll update this to match the current state. Thanks

celinval replied:
I documented that we are in a transition period. Let me know if that makes sense.

compiler-errors reviewed (changes requested):
Some nits. Let me know if I misunderstood something, particularly in the "Stable MIR design" section.

compiler-errors commented:
Anyways r=me after tweaking unless you specifically want a review from scott

celinval committed:
Apply suggestions from code review

Co-authored-by: Michael Goulet <michael@errs.io>

celinval committed:
Update README.md

Clarify that the translation between unstable and stable items is currently done in the `rustc_smir` crate.

celinval commented:
@compiler-errors do you mind taking a quick look at the updated version? Thanks

compiler-errors commented:
@bors r+ rollup
